### PR TITLE
feat: optimize plugin chunks

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-js-common/src/types/LoadOptions.ts [hunk 1]
```diff
@@ -127,4 +144,6 @@ export type LoadOptions = {
   dataPlaneEventsBufferTimeout?: number;
   storage?: StorageOpts;
   preConsent?: PreConsentOptions;
+  // transport mechanism to be used for sending batched requests
+  transportMode?: EventsTransportMode; // Unused for now. This will deprecate the useBeacon and beaconQueueOptions
```
Comment (bardisg) [inline packages/analytics-js-common/src/types/LoadOptions.ts:148]: Do we need to add a breaking change for this in the docs?

File: packages/analytics-js-plugins/package.json [hunk 2]
```diff
@@ -73,6 +73,7 @@
     "@preact/signals-core": "1.4.0",
     "@rudderstack/analytics-js-common": "3.0.0-beta.9",
     "crypto-js": "4.1.1",
```
Comment (bardisg) [inline packages/analytics-js-plugins/package.json:75]: can we now remove this one?

File: packages/analytics-js-common/src/types/LoadOptions.ts (hunk 1 above)
Comment (saikumarrs) [reply packages/analytics-js-common/src/types/LoadOptions.ts:148]: This option is not implemented yet in the code. So, no changes required.

File: packages/analytics-js-plugins/package.json (hunk 2 above)
Comment (saikumarrs) [reply packages/analytics-js-plugins/package.json:75]: This package is still used in the v1.1 package. We can remove it once we upgrade that too later.

Comment (bardisg) [reply packages/analytics-js-plugins/package.json:75]: but this is in the package json of the plugins that is not relevant to v1.1

**Pull request merged**